Solution Implemented for Emoji Rendering in Messages (#15714)

sumit2004tiwari opened the pull request:
<!--
Thank you for your pull request. Please provide a thorough description below.

Dear Sir/mam,

I have successfully resolved the issue related to rendering emojis while typing in messages (#15714). The implementation now ensures that text-based emoji inputs are automatically converted into their respective emoji representations.

Please review the changes at your convenience and let me know if any further modifications or improvements are required. I appreciate your feedback and look forward to your response.

Best regards,
Sumit Tiwari

jitsi-jenkins commented:
Hi, thanks for your contribution!
If you haven't already done so, could you please make sure you sign our CLA (https://jitsi.org/icla for individuals and https://jitsi.org/ccla for corporations)? We would unfortunately be unable to merge your patch unless we have that piece :(.

sumit2004tiwari commented:
# Code Update & Lint Issue Fix  (Solution Implemented for Emoji Rendering in Messages (#15714) #15791)

Hi @damencho,

I have updated my code and resolved the linting issues. I believe it should be working fine now. Could you please review it when you have a moment? Looking forward to your feedback.

Thanks!

Best regards,
Sumit Tiwari

manzil-infinity180 commented on react/features/chat/components/web/ChatInput.tsx:
```diff
@@ -274,12 +298,22 @@ class ChatInput extends Component<IProps, IState> {
     return {
         _areSmileysDisabled: areSmileysDisabled(state),
-        _privateMessageRecipientId: privateMessageRecipient?.id
+        _privateMessageRecipientId: privateMessageRecipient?.id,
```
you have added extra commas here

sumit2004tiwari replied:
yes sir i fix it . and i also run lint cmd and there is no issue i found  .
Sir we can cleary see that there is no lint error in local . can you please check it once .
![Screenshot 2025-03-16 092658](https://github.com/user-attachments/assets/dedc893e-76a1-431d-ac84-0cb1f82241c7)

manzil-infinity180 commented:
please run `npm run lint` locally and see what are the lint error their and fix it or run `npm run lint --fix` - thanks

manzil-infinity180 commented on react/features/chat/components/web/ChatInput.tsx:
```diff
@@ -59,12 +60,39 @@ interface IState {
+/**
+ * Custom Emoji Mapping for Replacement.
+ */
+const emojiMap: Record<string, string> = {
+    ':)': '😊',
+    ':(': '😦',
+    ':D': '😃',
+    ':+1:': '👍',
+    ':P': '😛',
+    ':wave:': '👋',
+    ':blush:': '😊',
+    ':slightly_smiling_face:': '🙂',
+    ':scream:': '😱',
+    ':*': '😗',
+    ':-1:': '👎',
+    ':mag:': '🔍',
+    ':heart:': '❤️',
+    ':innocent:': '😇',
+    ':angry:': '😠',
+    ':angel:': '👼',
+    ';(': '😭',
+    ':clap:': '👏',
+    ';)': '😉',
+    ':beer:': '🍺'
+};
```
as from @saghul comment , i think we are using some library for this which is `react-emoji-render`

sumit2004tiwari replied:
@manzil-infinity180  no one tell to use any library . i just check it . if you say i can implement library instead of json . but there is no boundation from saghul  sir  .

sumit2004tiwari commented on react/features/welcome/components/WelcomePage.web.tsx:
```diff
@@ -248,7 +248,7 @@ class WelcomePage extends AbstractWelcomePage<IProps> {
                                 <button
                                     aria-disabled = 'false'
-                                    aria-label = 'Start meeting'
+                                    aria-label = 'Start meeting123'
```
i revert my changes . but this line is not shown in my system locally 
![Screenshot 2025-03-16 092136](https://github.com/user-attachments/assets/0eaab689-6548-4ee3-ab70-9928c430616a)
.  i think it will resolve now

ahmedasad236 commented:
@manzil-infinity180 
I have already added this feature in this [PR](https://github.com/jitsi/jitsi-meet/pull/15715) since 2 weeks, and I requested to be reviewd.

sumit2004tiwari commented:
### Code Review Request – Lint Fixes, Emoji Rendering & Copy Message Functionality
Hello @manzil-infinity180 sir , @jitsi-jenkins 

I have resolved all lint issues, implemented emoji rendering using react-render-emoji, and added the copy message functionality. Could you please review my code and provide your feedback? . if you want any assistance please inform me .

I am eagerly awaiting your response and hope everything is working fine this time. Looking forward to your valuable insights.

Best regards,
Sumit Tiwari
